pkg/splunk/enterprise/configuration.go

```diff
@@ -516,9 +544,10 @@ func getSplunkStatefulSet(ctx context.Context, client splcommon.ControllerClient
 				Annotations: annotations,
 			},
 			Spec: corev1.PodSpec{
-				Affinity:      affinity,
-				Tolerations:   spec.Tolerations,
-				SchedulerName: spec.SchedulerName,
+				Affinity:         affinity,
+				Tolerations:      spec.Tolerations,
+				SchedulerName:    spec.SchedulerName,
+				ImagePullSecrets: spec.ImagePullSecrets,
```

Comment:
Can we not populate, when the list is empty?

Reply:
If `spec.ImagePullSecrets` is empty, it assigns an empty value which the StatefulSet will do by default?

Reply:
After thinking through, empty slice is required, considering the spec update cases, we are good here.